Fix for color progress in challenge userprogress modal fixes issue #10907

gautamz07 opened the pull request:
Fix for color progress in challenge userprogress modal fixes issue #10907

### Fixes

Fixes #10907

### Changes
basically passing a prop from `challengeMemberProgressModal.vue ` , the prop being `:showtaskcolor='true' `

@Alys have not added any tests, as i've never written any , will this be required ? please let me know :)

gautamz07 committed:
Fix for color progress in challenge userprogress modal

Fix for color progress in challenge userprogress modal

paglias commented:
Hi @gautamz07  and thanks for the PR! Sorry for the delay here but we'll review the pr soon

paglias commented on website/client/components/tasks/column.vue:
```diff
@@ -306,6 +307,9 @@ export default {
       default: false,
     },
+    showtaskcolor : {
+      type: Boolean
```
can you give this a default value and rename it to showTaskColor? thanks!

paglias reviewed (commented):
I've left one comment, but otherwise it looks good. Regarding the tests it's fine if the client doesn't have them

paglias commented:
Ah, I forgot but it looks like there are some linting issues https://travis-ci.org/HabitRPG/habitica/jobs/552780174

Also, could you revert the changes to package-lock.json? (`git checkout master -- package-lock.json`)

thanks!

gautamz07 committed:
Updating fixes requested

- Package-lock.json updated from develop.
- showTaskColor spell correction done.

gautamz07 commented:
@paglias Thanks alot for the inputs , changes requested have been made , check my comments for details.

Just curious , how did the package-lock.json file change even through i did't change anything in it ?

Also, the coverall's is failing, any idea why ?

gautamz07 committed:
Linting correction

Linting corrections.

paglias commented:
@gautamz07 thanks for the updates! I'll give it a final test later this week but it looks good

> Just curious , how did the package-lock.json file change even through i did't change anything in it ?

Eh it happens sometimes when you run `npm install` due to different operative systems / npm versions but it should only happen right after you install habitica for the first time.

> Also, the coverall's is failing, any idea why ?

to be honest coverall's being a bit buggy lately so don't worry about it, the important is that travis is green :)

gautamz07 commented:
> to be honest coverall's being a bit buggy lately so don't worry about it, the important is that travis is green :)

Thanks that will be great :) :+1:

gautamz07 commented:
@paglias any update on this ?

paglias commented:
Sorry for the delay! I just tested this and it's removed the gray state from user tasks as well 
<img width="352" alt="Schermata 2019-08-18 alle 17 08 50" src="https://user-images.githubusercontent.com/589911/63226532-ede8d080-c1da-11e9-84de-fdc4dbff0773.png">

I think the problem is because the default value is `true` and it's never set to false. I think it might be better to change the name to `alwaysShowTaskColor`, set the default value to false and then set it to true only for the challenge progress. Does it makes sense?

paglias commented:
Hi @gautamz07 , do you have any update on this PR? thanks!

paglias commented:
Hi @gautamz07 , I'm going to close this one but if you want to work again on this just leave a comment
